pkg/backends/calico/routes.go

```diff
@@ -348,6 +348,12 @@ func contains(items []string, target string) bool {
 func (rg *routeGenerator) advertiseThisService(svc *v1.Service, ep *v1.Endpoints) bool {
 	logc := log.WithField("svc", fmt.Sprintf("%s/%s", svc.Namespace, svc.Name))
 
+	// Don't advertise routes if this node is explicitly excluded from load balancers.
+	if rg.client.ExcludeServiceAdvertisement() {
+		logc.Debug("Skipping service because node is explicitly excluded from load balancers")
+		return false
+	}
```

Comment:
I wonder if this aspect of the change is useful.  If a node has been labelled to exclude, but it *has* a backend pod for a service with externalTrafficPolicy Local, I think we should still advertise the /32 service IP from this node.  Otherwise the service might not be accessible at all - e.g. if it has no other backend.

(The RFE customer shouldn't care, because in their setup they *won't* have any service backend pods on the labelled nodes.)

Reply:
Yeah, good point. I'm not sure what the best behavior is for when nodes have the "exclude-balancer" label set, but are also running a pod with `externalTrafficPolicy: Local` - it feels like a conflicting set of instructions.

On the one hand, I guess the label doesn't signify "Don't send any external traffic to this node" just "Don't use this node as a LB and route traffic to other destinations through this node", so I think the way you're describing is probably right.

But I can also imagine someone wanting to make sure a node doesn't get external traffic sent to it and then accidentally scheduling a pod on it due to a misconfig, in which case I think the expected behavior would be to not advertise that route.

Reply:
I'm going to merge this as-is for now while I continue to ponder this. I think it's OK if this ends up being the result, since it's easier to move from the way it is currently to a more permissive approach than it is vice-versa.